### ISSUE TITLE
Routing package from valhalla tiles

### ISSUE THREAD

**Issue by MJokar67** [CartoDB/mobile-sdk#552]
Hi 
I want to know how to create a routing package .vtiles from Valhalla tiles. As you know generated tiles from Valhalla are tailored as a special file name hierarchy like 'valhalla_tiles/1/001/002/0.gph'.
how to create an SQLite file and save every tile data as zoom_level, tile_row, tile_column, and tile_data?

**Comment by farfromrefug** [comment]
@MJokar67 I have an updated version of the build script for vtiles. See this https://github.com/Akylas/alpimaps_data_generator?tab=readme-ov-file#-build-valhalla-package.
BTW the framework is not maintained anymore. I am starting an effort to maintain a fork.
https://github.com/Akylas/mobile-sdk
https://github.com/Akylas/mobile-sdk/issues/1

**Comment by MJokar67** [comment]
> @MJokar67 I have an updated version of the build script for vtiles. See this https://github.com/Akylas/alpimaps_data_generator?tab=readme-ov-file#-build-valhalla-package. BTW the framework is not maintained anymore. I am starting an effort to maintain a fork. https://github.com/Akylas/mobile-sdk [Akylas#1](https://github.com/Akylas/mobile-sdk/issues/1)

Thank you very much for your kindly guidance. 
Is poly data mandatory here or I just use the pbf enough ?

**Comment by farfromrefug** [comment]
@MJokar67 no.it is not you should be able to.omit it

**Comment by MJokar67** [comment]
> @MJokar67 I have an updated version of the build script for vtiles. See this https://github.com/Akylas/alpimaps_data_generator?tab=readme-ov-file#-build-valhalla-package. BTW the framework is not maintained anymore. I am starting an effort to maintain a fork. https://github.com/Akylas/mobile-sdk [Akylas#1](https://github.com/Akylas/mobile-sdk/issues/1)

I've successfully generate routing tile package with your provided scrips. Thank you.
But there is some crash over new routing types its because of using latest provided valhalla tiles with the original  outdated Carto SDK. 

It seems i must recompile valhalla from source with the same version of Carto-SDK you are provided.

Do you have anything's for offline Geo coding package generation?

**Comment by farfromrefug** [comment]
@MJokar67 you can also try to use the build of the carto sdk from my fork
As for the geocoding thé reamd i shared before should mention how to generate it?

**Comment by MJokar67** [comment]
> @MJokar67 you can also try to use the build of the carto sdk from my fork As for the geocoding thé reamd i shared before should mention how to generate it?

Yes please🙏

**Comment by farfromrefug** [comment]
Here you go https://github.com/nutiteq/mobile-sdk-scripts?tab=readme-ov-file#creating-offline-geocoding-packages

**Comment by MJokar67** [comment]
Thank you.

**Comment by MJokar67** [comment]
> Here you go https://github.com/nutiteq/mobile-sdk-scripts?tab=readme-ov-file#creating-offline-geocoding-packages

hi bro.
Thank you in advance for your best works.

I mixed up your changes with the original version, but a few things occurred to me.

Some features of the code have been removed in your version, for example, the offline and online sections of Vector Tiles, package manager and ....
Culling part of labels have been replaced by another class, and I don't know about the new class performance.

I saw you are commented the blending with off screen buffer but at this moment I don't have any problem with uncommented version.

As you said, the rooting section was updated and thank you for that, but the changes I saw in the sources worried me, and that is whether your version is faster or slower than the original version?

Did you apply your changes to the latest version of the original SDK?
